dev/BBimie/misclassification_function.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+
+# In[ ]:
+
+
+from sklearn.model_selection import StratifiedShuffleSplit
+from sklearn.ensemble import RandomForestClassifier
+
+
+# In[ ]:
+
+
+clf = RandomForestClassifier()
+
+
+# This function takes in the test set, the target and the prediction and returns the total length of the test set and how many datapoints were missclassified
+
+# In[ ]:
+
+
+def misclassified_data(df, target, prediction):
+    misclassified = df[df[target] != df[prediction]]
+    print('Total Test Set: {}'.format(len(df)))
+    print("Number of Misclassified Datapoints: {}".format(len(misclassified)))
+
+
+# In[ ]:
+
+
+def plot_misclassified(df, target):
+    #stratified shuffle split because the data is not balanced
+    split = StratifiedShuffleSplit(n_splits = 1, test_size=0.2, random_state=42)
+    for train_index, test_index in split.split(df, df[target]):
+        train = df.loc[train_index]
+        test = df.loc[test_index]
+        
+        #train feature
+        x = train.drop([target], axis=1)
+        #train target
+        y = train[target]
+    
+        #test feature
+        a = test.drop([target], axis=1)
+        #test target
+        b = test[target]
+    
+        #fit the random forest model
+        clf.fit(x,y)
```

Comment:
Rather than doing train/test splits and fitting the model inside this function, it would be nicer to have a dedicated function for computing the probabilities and creating the visualization that take as input a fitted model and dataset.